llvm/lib/Target/EVM/EVMOptimizedCodeTransform.cpp

```diff
@@ -122,12 +122,17 @@ EVMOptimizedCodeTransform::EVMOptimizedCodeTransform(EVMAssembly &Assembly,
                                                      MachineFunction &MF)
     : Assembly(Assembly), Layout(Layout), FuncInfo(&Cfg.FuncInfo), MF(MF) {}
 
-void EVMOptimizedCodeTransform::assertLayoutCompatibility(
-    Stack const &SourceStack, Stack const &TargetStack) {
-  assert(SourceStack.size() == TargetStack.size());
+bool EVMOptimizedCodeTransform::AreLayoutsCompatible(Stack const &SourceStack,
+                                                     Stack const &TargetStack) {
+  if (SourceStack.size() != TargetStack.size())
+    return false;
+
```

Comment:
The remainder can be folded to an algorithm (all / any)

Reply:
Thank you for pointing out, I've used `llvm::zip_equal`.

Reply:
And you're one step away from `llvm::any`